### ISSUE TITLE
[vm] Fail to build with `-Werror=format-security` enabled

### ISSUE THREAD

**Issue by otavio** [TotalCross/totalcross#31]
Using `-Werror=format-security` on GCC to increase security on the generated code, the build fails.

For reference: https://fedoraproject.org/wiki/Format-Security-FAQ

**Comment by ItaloYeltsin** [comment]
We have a lot of warnings to solve in order to make this build possible using such a flag. For example, for iOS, the number of warnings is about 460.

**Comment by otavio** [comment]
Yeah, we disabled it for now but those warnings are real issues and it'd be good to eventually fix them.

**Comment by acmlira** [comment]
In the next few days we will be working on things nearby #71 we can resume this discussion.
